GameDataParser/Files/Hash.cs

```diff
@@ -6,53 +6,40 @@ namespace GameDataParser.Files
 {
     public static class Hash
     {
-        public static bool CheckHash(string filename)
+        public static bool HasValidHash(string filename)
         {
-            try
-            {
-                string currentHash = File.ReadAllText($"{Paths.HASH}/{filename}-hash");
-
-                using (MD5 md5 = MD5.Create())
-                {
-                    using (FileStream stream = File.OpenRead($"{Paths.OUTPUT}/{filename}"))
-                    {
-                        byte[] hash = md5.ComputeHash(stream);
-                        string newHash = BitConverter.ToString(hash).Replace("-", "").ToLowerInvariant();
+            string hashPath = $"{Paths.HASH}/{filename}-hash";
 
-                        return currentHash.Equals(newHash);
-                    }
-                }
-            }
-            catch (Exception ex)
+            if (!File.Exists(hashPath))
             {
-                if (!ex.GetType().IsAssignableFrom(typeof(FileNotFoundException)))
-                {
-                    Console.WriteLine(ex);
-                }
-
                 return false;
             }
+
+            string currentHash = File.ReadAllText(hashPath);
+            string newHash = GetHash(filename);
+
+            return currentHash.Equals(newHash);
         }
 
         public static void WriteHash(string filename)
         {
-            try
+            string hashPath = $"{Paths.HASH}/{filename}-hash";
+
+            string newHash = GetHash(filename);
+
+            File.WriteAllText(hashPath, newHash);
+        }
+
+        public static String GetHash(string filename)
+        {
+            using (MD5 md5 = MD5.Create())
             {
-                using (MD5 md5 = MD5.Create())
+                using (FileStream stream = File.OpenRead(Paths.OUTPUT + "/" + filename))
```

Comment:
I feel that the use of string interpolation here originally was more readable than changing it to use concatenation. It works either way, but I think the change was unnecessary. It's also inconsistent with the other paths created in this same file now. I recommend changing this one back, or changing all other instances to use concatenation.

Reply:
Updated